src/ResourceManager/ApiManagement/Commands.ApiManagement/Commands/SetAzureApiManagement.cs

```diff
@@ -0,0 +1,58 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.Models;
+    using Properties;
+    using ResourceManager.Common.ArgumentCompleters;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.Set, "AzureRmApiManagement", SupportsShouldProcess = true), OutputType(typeof(PsApiManagement))]
+    public class SetAzureApiManagement : AzureApiManagementCmdletBase
+    {
+        [Parameter(
+            ValueFromPipeline = true,
+            Mandatory = true,
+            HelpMessage = "The ApiManagement instance.")]
+        [ValidateNotNull]
+        public PsApiManagement ApiManagement { get; set; }
```

Comment:
@solankisamir also, we should look at adding another parameter set containing the identity properties of an `ApiManagement` object that the user could provide to this cmdlet so they don't always have to provide the `ApiManagement` object

Reply:
I will take it as feedback for the next update to the commandlet.

---
In reply to: [188127907](https://github.com/Azure/azure-powershell/pull/6189#discussion_r188127907) [](ancestors = 188127907)